### PR TITLE
bump pkg(5) dependencies for Python 3.9

### PR DESCRIPTION
**Pull request by richlowe**
This builds pycurl for python 3.9 too, which would be necessary to use python 3.9 in pkg.

Problems:
- The build can't tell curl is built to use OpenSSL unless I tell it so, and I dont know why.
- `make test` doesn't work, and seems like it never has?  

I've marked this draft to make it absolutely clear I have no idea if it even works.

I've also bumped (and am in the process of bumping) everything else pkg(5) would need

### CODE REVIEW

**Comment by richlowe** [comment]
These should all probably use `$(PYTHON_VERSIONS_ALL)` but don't right now

**Comment by richlowe** [comment]
These now use `$(PYTHON_VERSIONS_ALL)` etc as appropriate, and seem to work well enough to run the pkg tests under both 3.5 and 3.9 (with a pkg modified for the purpose, which comes next).

File: components/python/jsonrpclib/Makefile
Comment (AndWac) [inline components/python/jsonrpclib/Makefile]: If you change the COMPONENT_VERSION you should reset the COMPONENT_REVISION, eg. by removing it.

File: components/python/cheroot/Makefile
Comment (AndWac) [inline components/python/cheroot/Makefile:23]: This is a jump from COMPONENT_REVISION=0 (not set) to 2.

File: components/python/pycurl/Makefile
Comment (AndWac) [inline components/python/pycurl/Makefile:31]: COMPONENT_REVISION can be removed as COMPONENT_VERSION has changed.
As long as we don't release a new package version the COMPONENT_REVISION only needs to be updated once.

**Comment by richlowe** [comment]
thanks @AndWac.  Now all I have to do is figure out the libbe flag day!

**Comment by AndWac** [comment]
I think we can remove the draft label and merge it, can't we?

File: components/web/apache2-modules/mod_wsgi/Makefile
Comment (AndWac) [inline components/web/apache2-modules/mod_wsgi/Makefile:30]: COMPONENT_REVISION can be removed.

**Comment by richlowe** [comment]
I'm honestly worried about whether it's been tested enough, but I can't work out better things to do.

**Comment by AndWac** [comment]
I will test the packages with version changes within the next days on my machine.
I don't think that we'll need to test packages where you only added python versions.

**Comment by AndWac** [comment]
My first test failed. After updating all packages I got
╰─➤  pkg list -a
Traceback (most recent call last):
  File "/usr/bin/pkg", line 87, in <module>
    import pkg.actions as actions
  File "/usr/lib/python3.5/vendor-packages/pkg/actions/__init__.py", line 68, in <module>
    globals(), locals(), [modname])
  File "/usr/lib/python3.5/vendor-packages/pkg/actions/directory.py", line 33, in <module>
    from . import generic
  File "/usr/lib/python3.5/vendor-packages/pkg/actions/generic.py", line 49, in <module>
    import pkg.variant as variant
  File "/usr/lib/python3.5/vendor-packages/pkg/variant.py", line 36, in <module>
    from pkg.misc import EmptyI
  File "/usr/lib/python3.5/vendor-packages/pkg/misc.py", line 76, in <module>
    import pkg.json as json
  File "/usr/lib/python3.5/vendor-packages/pkg/json.py", line 25, in <module>
    from jsonschema import validate, ValidationError
  File "/usr/lib/python3.5/vendor-packages/jsonschema/__init__.py", line 21, in <module>
    from jsonschema._types import TypeChecker
  File "/usr/lib/python3.5/vendor-packages/jsonschema/_types.py", line 3, in <module>
    from pyrsistent import pmap
ImportError: No module named 'pyrsistent'

So we need to make sure that the corresponding pyrsistent packages will be installed before we merge this.
This also means that pyrsistent needs to be in a separate PR.

**Comment by AndWac** [comment]
When pyrsistent is installed before updating the packages pkg seem to continue to work.

File: components/web/apache2-modules/mod_wsgi/pkg5
Comment (AndWac) [inline components/web/apache2-modules/mod_wsgi/pkg5:17]: "web/server/apache-24/module/apache-wsgi-37",
is missing here.

**Comment by AndWac** [comment]
pyrsistent is missing its pkg5 file

**Comment by richlowe** [comment]
We should make jsonschema depend on pyrsistent.  But jsonschema is delivered for -27 whereas pyrsistent is 3 only.  What can I do there?

**Comment by AndWac** [comment]
Adding this in the jsonschema-PYVER.p5m file should do:

\# force a dependency on pyrsistent for python-3.x versions:
$(PYTHON_2.7_EXCL)depend type=require fmri=library/python/pyrsistent-$(PYV)

Nevertheless this dependency means that pyrsistent has to be merged first and thus needs a separate PR.

File: components/python/pyrsistent/pyrsistent-PYVER.p5m
Comment (AndWac) [inline components/python/pyrsistent/pyrsistent-PYVER.p5m]: You should add the standard runtime dependencies for python packages here:

\# force a dependency on the Python runtime
depend type=require fmri=__TBD pkg.debug.depend.file=python$(PYVER) \
    pkg.debug.depend.path=usr/bin

\# force a dependency on the pyrsistent package
depend type=require \
    fmri=library/python/pyrsistent@$(IPS_COMPONENT_VERSION),$(BUILD_VERSION)

**Comment by richlowe** [comment]
I _think_ the new jsonschema doesn't actually support 2.7 anymore.  Do we obselete it for that version? Do we... do something else?

**Comment by AndWac** [comment]
I have checked for dependencies on library/python/jsonschema-27 and the only result was library/python-2/jsonschema-27.
So we can obsolete this package in the local history file. But beware that this history file contains an entry that forwards the old library/python-2/jsonschema-27 to the newer library/python/jsonschema-27 package. This forwarding must be removed (rightmost column in that line has to be deleted) otherwise updates for it will be blocked until manually solved.

**Comment by richlowe** [comment]
Is this update ok except for splitting into a separate PR?
I can't test the obsoletion because I already have a newer version installed from doing other tests.

**Comment by AndWac** [comment]
pyrsistent is missing its generic-manifest file. You should add it as it is more important for python packages than the sample-manifest.
I will check everything again when you have split it into two PR's.

**Comment by richlowe** [comment]
See #7368.  Sorry I'm so bad at this

**Comment by Toasterson** [comment]
@richlowe Knowing how much manual work we have in the system that changes on a constant basis, it's no problem at all. Specially python is complex. We are here to help. We need to change some things but right now the best we can do is help. And we are through this big work soon, this will be a huge help. Thank you!

File: components/python/jsonschema/history
Comment (AndWac) [inline components/python/jsonschema/history]: In order to obsolete a package you'll need to increment the COMPONENT_REVISION
library/python/jsonschema-27@2.4.0-2020.0.1.6  ->  library/python/jsonschema-27@2.4.0-2020.0.1.7

**Review by AndWac — APPROVED** [review]

**Pull request merged**
